src/registers/ich_lr0_el2.rs

```diff
@@ -9,7 +9,48 @@
 //!
 //! Provides interrupt context information for the virtual CPU interface.
 
-use tock_registers::interfaces::{Readable, Writeable};
+use tock_registers::{
+    interfaces::{Readable, Writeable},
+    register_bitfields,
+};
+
+register_bitfields! {u64,
+    pub ICH_LR0_EL2 [
+        /// The state of the interrupt.
+        State OFFSET(62) NUMBITS(2) [
+            Invalid = 0b00,
+            Pending = 0b01,
+            Active = 0b10,
+            PendingAndActive = 0b11,
+        ],
+
+        /// Indicates whether this virtual interrupt maps directly to a hardware interrupt, meaning
+        /// that it corresponds to a physical interrupt. Deactivation of the virtual interrupt also
+        /// causes the deactivation of the physical interrupt with the ID that the pINTID field
+        /// indicates.
+        HW OFFSET(61) NUMBITS(1) [],
+
+        /// Indicates the group for this virtual interrupt.
+        Group OFFSET(60) NUMBITS(1) [],
+
+        /// When FEAT_GICv3_NMI is implemented:
+        /// Indicates whether the virtual priority has the non-maskable property.
+        NMI OFFSET(59) NUMBITS(1) [],
```

Comment:
Hmm, I don't see NMI field listed in the [ARM](https://developer.arm.com/documentation/ddi0595/2020-12/AArch64-Registers/ICH-LR-n--EL2--Interrupt-Controller-List-Registers?lang=en)

Reply:
I think we may need to agree on what version of the ARM we follow. This field does appear at least in Issue H of the Arm Generic Interrupt Controller (Arm IHI 0069H (ID020922)).

Reply:
I am good with supported the broadest range, and it already has a note "When FEAT_GICv3_NMI is implemented" so I guess that's good.